server/routing/pubsub/manager.go

```diff
@@ -0,0 +1,282 @@
+// Copyright AGNTCY Contributors (https://github.com/agntcy)
+// SPDX-License-Identifier: Apache-2.0
+
+package pubsub
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/agntcy/dir/server/types/labels"
+	"github.com/agntcy/dir/utils/logging"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	"github.com/libp2p/go-libp2p/core/host"
+)
+
+var logger = logging.Logger("routing/pubsub")
+
+// Manager handles GossipSub operations for label announcements.
+// It provides efficient label propagation across the network without
+// requiring peers to pull entire records.
+//
+// Architecture:
+//   - Publisher: Announces labels when storing records
+//   - Subscriber: Receives and caches labels from remote peers
+//   - Integration: Works alongside DHT for resilient discovery
+//
+// Performance:
+//   - Propagation: ~5-20ms (vs DHT's ~100-500ms)
+//   - Bandwidth: ~100B per announcement (vs KB-MB for full record pull)
+//   - Reach: ALL subscribed peers (vs DHT's k-closest peers)
+type Manager struct {
+	ctx         context.Context //nolint:containedctx // Needed for long-running message handler goroutine
+	host        host.Host
+	pubsub      *pubsub.PubSub
+	topic       *pubsub.Topic
+	sub         *pubsub.Subscription
+	localPeerID string
+	topicName   string // Topic name (protocol constant)
+
+	// Callback invoked when label announcement is received
+	onLabelAnnouncement func(context.Context, *LabelAnnouncement)
```

Comment:
I feel like we are doing this the other way. Instead of doing `label` announcement, its actually `record` announcement with extra data, one of which is labels themselves.

IMO this would be clearer if we had `PublishEvent` as a general publication event, or even more specifically, `RecordPublishEvent`

 
```suggestion
	onRecordAnnouncement func(context.Context, *RecordPublishEvent)
```